app/actions/views/channel.js

```diff
@@ -687,6 +691,80 @@ export function loadChannelsForTeam(teamId) {
     };
 }
 
+export function preFetchUnreadChannels(unreadChannels) {
+    return async (dispatch, getState) => {
+        const state = getState();
+        const {posts} = state.entities.posts;
+        const time = Date.now();
+
+        const channelFetchPromises = unreadChannels.map(async (channelId) => {
+            const postsIds = getPostIdsInChannel(state, channelId);
+            const actions = [];
+            let newPosts;
+            if (!postsIds || postsIds.length < ViewTypes.POST_VISIBILITY_CHUNK_SIZE) {
+                // Get the first page of posts if it appears we haven't gotten it yet, like the webapp
+                for (let i = 0; i <= MAX_RETRIES; i++) {
+                    try {
+                        newPosts = await Client4.getPosts(channelId, 0, 60); //eslint-disable-line no-await-in-loop
+                        getProfilesAndStatusesForPosts(newPosts.posts, dispatch, getState);
+                        actions.push(
+                            receivedPosts(newPosts),
+                            receivedPostsInChannel(newPosts, channelId, true, newPosts.prev_post_id === ''),
+                        );
+                        break;
+                    } catch (err) {
+                        return {err};
+                    }
+                }
+            } else {
+                const lastConnectAt = state.websocket?.lastConnectAt || 0;
+                const lastGetPosts = state.views.channel.lastGetPosts[channelId];
+
+                let since;
+                if (lastGetPosts && lastGetPosts < lastConnectAt) {
+                    // Since the websocket disconnected, we may have missed some posts since then
+                    since = lastGetPosts;
+                } else {
+                    // Trust that we've received all posts since the last time the websocket disconnected
+                    // so just get any that have changed since the latest one we've received
+                    const postsForChannel = postsIds.map((id) => posts[id]);
+                    since = getLastCreateAt(postsForChannel);
+                }
+
+                for (let i = 0; i <= MAX_RETRIES; i++) {
+                    try {
+                        newPosts = await Client4.getPostsSince(channelId, since); //eslint-disable-line no-await-in-loop
+                        getProfilesAndStatusesForPosts(newPosts.posts, dispatch, getState);
+                        actions.push(
+                            receivedPosts(newPosts),
+                            receivedPostsSince(newPosts, channelId),
+                            {
+                                type: PostTypes.GET_POSTS_SINCE_SUCCESS,
+                            },
+                        );
+                        break;
+                    } catch (err) {
+                        return {err};
+                    }
+                }
+            }
```

Comment:
This seems to be duplicating a lot of code, maybe this can be done better?

Also you are dispatching the same action with different values a few times here, maybe best to consider a different action that takes all posts and then you'll only need one dispatch?

for `getProfilesAndStatusesForPosts` the same can be applied where instead of calling it multiple times for each channel you combine all posts and call it just once?